src/main/java/Transcription/CacheKey.java

```diff
@@ -1,21 +1,18 @@
-package Search;
-
-import SpeechToText.SpeechToText;
+package Transcription;
 
 import java.io.File;
 
 public class CacheKey {
-
     private static final String CACHE_DIRECTORY = ".cache";
     private static final String TRANSCRIPT_FILE_EXTENSION = ".transcript";
     private static final String TIMESTAMP_FILE_EXTENSION = ".timestamp";
 
     public String filename;
     public String lastModified;
-    public SpeechToText speechToText;
+    public VoskAdapter speechToText;
     private String hash;
 
-    public CacheKey(String filename, SpeechToText speechToText) {
+    public CacheKey(String filename, VoskAdapter speechToText) {
```

Comment:
This object has too much functionality. Make a separate CacheKey that is only used by the Cache, and then have a different DTO for passing file information from the cache to the other classes.

Reply:
Made CacheInfo object, TranscriptionCache creates this object and passes info to searchable and VoskAdapter.